src/pybind/mgr/rbd_support/schedule.py

```diff
@@ -507,7 +507,7 @@ def remove(self,
                level_spec: LevelSpec,
                interval: Optional[str],
                start_time: Optional[str]) -> None:
-        schedule = self.schedules.pop(level_spec.id, None)
+        schedule = self.schedules.get(level_spec.id, None)
```

Comment:
And now what about the case when you remove a valid interval? `self.schedules[level_spec.id]` will contain the empty schedule. It may be not an issue (would need to check) but does not look nice.

Note on the line 518 we reinsert the schedule back only if it is not empty, while after your change it will remain there even if it is empty.

I suppose the problem is that `Interval.from_string(interval)`  below throws `ValueError` exception for an invalid interval, and it skips the code that reinserts the scheduler back.

So we need either to catch the ValueError` exception and proceed with insertion, or even better validate the interval earlier and return a meaningful error message to the user.

Reply:
+1 for validation checking

Reply:
done